### DISCUSSION TITLE
Unable to build webpack sucessfully for test automation

### DISCUSSION THREAD

**Discussion by srikanthathikari** [webpack/webpack#16166] [Q&A]
**Do you want to request a *feature* or report a *bug*?** - Request a feature

**What is the current behavior?**
I am currently using playwright which is a test automation framework. I have end-to-end tests that have long scenario tests. I am trying to split them into multiple files and build them using webpack at run time and run the long scenario from multiple files to a single test. I am not 100% sure if this is something possible. 

**If the current behavior is a bug, please provide the steps to reproduce.**

 **https://github.com/srikanthathikari/Playwright/tree/feat-webpack**

**What is the expected behavior?**

**If this is a feature request, what is the motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

**Comment by vankop** [comment]
Could you describe your problem? webpack config looks correct ( note you have `exclude: /node_modules/` which means node_modules code could be incompatible with your browser )

**Reply by srikanthathikari** [reply]
Thank you for the reply. While I am trying to build with webpack, I get the error mentioned in the screenshot. 

![image](https://user-images.githubusercontent.com/25626394/185646155-ac83a2db-9e6f-4e61-a857-8011962e87f9.png)

**Reply by alexander-akait** [reply]
Looks like you don't have `general_helper` file, can you provide example of code?

**Reply by srikanthathikari** [reply]
https://github.com/srikanthathikari/Playwright/tree/feat-webpack
Please find the link to my code. The general helper is available. I tried removing the dependency of the general helper and I still get the error.

**Comment by srikanthathikari** [comment]
Also, I observed that whenever I build without the playwright imports the build is successful. If I add imports at top of the test file the build is failing. 

![image](https://user-images.githubusercontent.com/25626394/185688578-976744ec-5dc0-4402-ae0f-f1c9eabacaa3.png)

**Reply by alexander-akait** [reply]
Because `playwright` is Node.js package, but you try to bundle it for web, please set `target: 'node'`, anyway `playwright` have dynamic imports, so you need to ignore them using `noParser`

Also why do you bundle tests? I don't think you need it, just use ESM modules and run `playwright`
